src/libtorch.cc

```diff
@@ -755,12 +758,15 @@ ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
           input_index_map_[io_name] = ip_index;
         }
         catch (std::exception& ex) {
-          return TRITONSERVER_ErrorNew(
-              TRITONSERVER_ERROR_INTERNAL,
-              ("input '" + io_name +
-               "' is neither an input argument to the model nor does it "
-               "follow the naming convention i.e. <name>__<index>.")
+          LOG_MESSAGE(
+              TRITONSERVER_LOG_WARN,
+              ("input '" + io_name + "' is neither an input argument to the " +
+               "model '" + model_state_->Name() +
+               "' nor does it follow the naming convention i.e. "
+               "<name>__<index>. Falling back to enforcing ordering from model "
+               "configuration.")
                   .c_str());
+          input_index_map_[io_name] = i;
```

Comment:
Clarifying question: We're allowing users to potentially set a convention for some inputs/outputs but not others? I'm not sure if this is an intended side effect of this ticket, but wanted to confirm.

Reply:
I was aware of this side effect and added testing for it as well. If the consensus of the team is to force a specific convention we can follow that. That being said we do not strictly validate IO names. In fact one can load a model with 2 inputs both as INPUT__0 at the moment.

Reply:
Perfect, thanks for explaining and testing.